src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1149,15 +1177,860 @@ bool PhaseIterGVN::verify_node_value(Node* n) {
+    //   calls commute -> left is Load, right not -> commute.
+    //
+    // Not sure why notification does not work here, seems like
+    // the depth is only 1, so it should work. Needs investigation.
+    //
+    // Found with:
+    //   compiler/codegen/TestCharVect2.java#id0
+    //   -XX:VerifyIterativeGVN=1110
+    case Op_OrI:
+    case Op_OrL:
+      return false;
+
+    // Bool -> constant folded to 1.
+    // Issue with notification?
+    //
+    // Found with:
+    //   compiler/c2/irTests/TestVectorizationMismatchedAccess.java
+    //   -XX:VerifyIterativeGVN=1110
+    case Op_Bool:
+      return false;
+
+    // LShiftLNode::Ideal
+    // Looks at pattern: "(x + x) << c0", converts it to "x << (c0 + 1)"
+    // Probably a notification issue.
+    //
+    // Found with:
+    //   compiler/conversions/TestMoveConvI2LOrCastIIThruAddIs.java
+    //   -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    case Op_LShiftL:
+      return false;
+
+    // LShiftINode::Ideal
+    // pattern: ((x + con1) << con2) -> x << con2 + con1 << con2
+    // Could be issue with notification of inputs of inputs
+    //
+    // Side-note: should cases like these not be shared between
+    //            LShiftI and LShiftL?
+    //
+    // Found with:
+    //   compiler/escapeAnalysis/Test6689060.java
+    //   -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110 -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    case Op_LShiftI:
+      return false;
+
+    // AddPNode::Ideal seems to do set_req without removing lock first.
+    // Found with various vector tests tier1-tier3.
+    case Op_AddP:
+      return false;
+
+    // StrIndexOfNode::Ideal
+    // Found in tier1-3.
+    case Op_StrIndexOf:
+    case Op_StrIndexOfChar:
+      return false;
+
+    // StrEqualsNode::Identity
+    //
+    // Found (linux x64 only?) with:
+    //   serviceability/sa/ClhsdbThreadContext.java
+    //   -XX:+UnlockExperimentalVMOptions -XX:LockingMode=1 -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    case Op_StrEquals:
+      return false;
+
+    // AryEqNode::Ideal
+    // Not investigated. Reshapes itself and adds lots of nodes to the worklist.
+    //
+    // Found with:
+    //   vmTestbase/vm/mlvm/meth/stress/compiler/i2c_c2i/Test.java
+    //   -XX:+UnlockDiagnosticVMOptions -XX:-TieredCompilation -XX:+StressUnstableIfTraps -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    case Op_AryEq:
+      return false;
+
+    // MergeMemNode::Ideal
+    // Found in tier1-3.
+    case Op_MergeMem:
+      return false;
+
+    // URShiftINode::Ideal
+    // Found in tier1-3.
+    case Op_URShiftI:
+      return false;
+
+    // CMoveINode::Ideal
+    // Found in tier1-3.
+    case Op_CMoveI:
+      return false;
```

Comment:
Maybe merge them together and add a comment that you have not investigated further (I assume?) since you found them all in tier1-3 without more specific details.

Reply:
I would prefer to keep them separate, so it is easier to remove them individually without getting merge conflicts.

Reply:
I'll add a comment that I did not investigate further yet.